desc/integrals/_bounce_utils.py

```diff
@@ -779,26 +766,31 @@ def cubic_spline(T, f, Y, n0, n1, NFP=1, check=False):
         Second to last axis enumerates the polynomials that compose a particular
         spline.
     knots : jnp.ndarray
-        Shape (num transit * (Y - 1)).
+        Shape (num transit * Y).
         Knots of spline ``f``.
 
     """
-    knots = jnp.linspace(-1, 1, Y, endpoint=False)
-    # θ at uniformly spaced points along field line
-    theta = idct_non_uniform(knots, T.cheb[..., jnp.newaxis, :], T.Y).reshape(
-        *T.cheb.shape[:-2], T.X * Y  # num transit * num points
-    )
-    knots = jnp.ravel(
-        bijection_from_disc(knots, T.domain[0], T.domain[-1])
-        + (T.domain[-1] - T.domain[0]) * jnp.arange(T.X)[:, jnp.newaxis]
+    num_transit = T.X
+    zeta = jnp.linspace(-1, 1, Y, endpoint=False)
+    # θ at uniformly spaced points in ζ along field line
+    theta = idct_non_uniform(zeta, T.cheb[..., jnp.newaxis, :], T.Y)
+    zeta = bijection_from_disc(zeta, T.domain[0], T.domain[1])
+
+    # f at uniformly spaced points in ζ along field line
+    f = _irfft2_non_uniform(
+        theta,
+        zeta,
```

Comment:
Take 5 min tomorrow to use a uniform fft in zeta here so that we only need 1d non uniform transform